src/sign.ts

```diff
@@ -49,42 +71,73 @@ async function sign(): Promise<void> {
         if (e instanceof Error) {
             core.setFailed(e);
         } else {
-            core.setFailed('Unknown error during notation sign');
+            core.setFailed('unknown error during notation sign');
         }
     }
 }
 
 // setupPlugin sets up the Notation signing plugin.
 async function setupPlugin() {
     try {
-        // inputs from user
-        const plugin_url = core.getInput('plugin_url');
-        const plugin_checksum = core.getInput('plugin_checksum').toLowerCase();
-        console.log(`signing plugin url is ${plugin_url}`);
+        console.log(`input plugin_name is ${plugin_name}`);
+        console.log(`input plugin url is ${plugin_url}`);
+        console.log(`input plugin checksum is ${plugin_checksum}`);
+
+        // check if plugin is already installed
+        const notationPluginPath = path.join(getConfigHome(), `notation/plugins/${plugin_name}`);
+        if (checkPluginExistence(notationPluginPath)) {
+            console.log(`plugin ${plugin_name} is already installed`);
```

Comment:
What if a user wants to install the same plugin but different versions in different steps?

Reply:
Great question. Initially this was considered, but I couldn't find a scenario that needs it. Suppose a user wants to sign multiple times in a single github workflow, why would they want to use different plugin versions (of the same plugin) between these signing steps? (If they really want to do that, they could have two separate workflows to do so.)
One technical difficulty if we do check version:
`notation sign` command actually does not support plugin version. Since it's a dir, we can only have one plugin with name, for example, `notation-azure-kv` under `notation/plugins/azure-kv`.

Reply:
We might want to create an issue to track this.

One rare scenario can be: use different versions of plugins to produce different versions of signature for backward compatibility if there is a breaking change between those two versions.

Reply:
Created issue #32 to track this.